src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1235,10 +1296,42 @@ private IndexWriter WriterFactory(Directory d)
             if (d == null) throw new ArgumentNullException(nameof(d));
             var writer = new IndexWriter(d, FieldAnalyzer, false, IndexWriter.MaxFieldLength.UNLIMITED);
 
-            // clear out current scheduler and set the error logging one
-            using (writer.MergeScheduler) { }
-            writer.SetMergeScheduler(new ErrorLoggingConcurrentMergeScheduler(Name,
-                (s, e) => OnIndexingError(new IndexingErrorEventArgs(this, s, "-1", e))));
+            if (_directory is ExamineDirectory examineDirectory)
```

Comment:
This block seems like pretty much the same code duplication above. Can we remove the duplication?